pcap/pcap.go

```diff
@@ -430,9 +446,86 @@ func (p *Handle) SetBPFFilter(expr string) (err error) {
 	defer C.free(unsafe.Pointer(cexpr))
 
 	if -1 == C.pcap_compile(p.cptr, &bpf, cexpr, 1, C.bpf_u_int32(maskp)) {
+		return bpf, p.Error()
+	}
+
+	return bpf, nil
+}
+
+// CompileBPFFilter compiles and returns a BPF filter for the pcap handle.
+func (p *Handle) CompileBPFFilter(expr string) ([]BPFInstruction, error) {
+	bpf, err := p.compileBPFFilter(expr)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
move the pcap_freecode call up here as a defer?

I'm worried that a future coder might throw a return somewhere else in this function.

defer C.pcap_freecode(&bpf)